chia/rpc/wallet_rpc_api.py

```diff
@@ -841,30 +842,30 @@ async def create_offer_for_ids(self, request):
             )
         if success:
             return {
-                "offer": trade_record.offer.hex(),
+                "offer": encode_offer_bytes(Offer.from_bytes(trade_record.offer).compress()),
                 "trade_record": trade_record.to_json_dict_convenience(),
             }
         raise ValueError(error)
 
     async def get_offer_summary(self, request):
         assert self.service.wallet_state_manager is not None
         offer_hex: str = request["offer"]
-        offer = Offer.from_bytes(hexstr_to_bytes(offer_hex))
+        offer = try_offer_decompression(decode_offer_bytes(offer_hex))
```

Comment:
It might be nice to have a single function that takes the `bech32m` offer blob and turns it into an `Offer` object. You use this pattern several times.